troubleshooting/conflicts.md

```diff
@@ -0,0 +1,163 @@
+
+## An example
+
+To make it clear what's happening, let's walk through an example. If you already understand conflicts feel free to skip this section.
+
+Let's say you register a pregnancy in the UI. And then you notice that you got the LMP wrong, so you hit edit and quickly make the change.
+
+Meanwhile, sentinel notices that you registers a pregnancy, and before you have a chance to change the LMP, sets up a bunch of recurring messages, editing the document.
+
+Sentinel has now made a change to the first version of your document, and you're trying to also make a change to the first version. These conflict.
+
+## How to manage conflicts
+
+Conflicts will appear on the #downtime Slack channel, along with the other alerts. If a server you manage has a conflict, you should do the following:
+
+### Identify why conflicts are occuring
+
+Take a look at the conflicts view, at `https://yourserver/medic/_design/medic-conflicts/_view/conflicts`. Each entry in that view looks like this:
+```json
+{
+    "id": "1B7922A6-A6D6-C956-BBAE-DE5EB5A2E6C8",
+    "key": [
+        "1-82d9a42305a79e403d9eca6a9a9daae9"
+    ],
+    "value": null
+},
+```
+
+The `id` is the `_id` of the conflicting document, and the `key` is a list of conflicting `_rev`s.
+
+For each conflicting document, download that document, as well as all the `_rev`s indicated in the `key` above. To download a document with a specific `_rev`, pass the `rev` parameter.
+
+```sh
+curl 'https://yourserver/medic/1B7922A6-A6D6-C956-BBAE-DE5EB5A2E6C8' > doc.json
+curl 'https://yourserver/medic/1B7922A6-A6D6-C956-BBAE-DE5EB5A2E6C8?rev=1-82d9a42305a79e403d9eca6a9a9daae9' > doc-conflict1.json
+```
+
+Now you have all versions of the document, you can diff them and try to determine what went wrong.
+
+A common problem, for example, might be sentinel hitting a document really quickly between you creating it and editing it.
+
+A less common problem that requires some special attention, is UUID collisions (see below).
+
+If you get stuck feel free to escalate to a developer, who can take a look.
+
+### If appropriate, raise a bug
+
+If you determine (or just suspect) that the problem is caused by bad design, feel free to raise a bug to development. For example, historically read status has been stored against the document, which can easily cause conflicts if you create a document and then instantly view it with sentinel processing in the background.
+
+While conflicts are inevitable, we want to architect away from them as much as possible. Ideally tech leads would never have to resolve conflicts.
+
+### Regardless, resolve the conflicts
+
+Now that you've diagnosed the problem, and perhaps reported a bug, you should resolve the conflict. This is important, as conflicts can cause important changes to not be recognised.
```

Comment:
This one is tricky, as that is not strictly true, depending on the circumstance. I'll try to clarify it.